File: C#/Watchdog/ReadWatchdogConfig/ReadWatchdogConfig.cs
//-----------------------------------------------------------------------------
// ReadWatchdogConfig.cs
//
// Demonstrates how to read the Watchdog configuration from a LabJack.
//
// support@labjack.com
//-----------------------------------------------------------------------------
using System;
using LabJack;

namespace ReadWatchdogConfig
{
    class ReadWatchdogConfig
    {
        static void Main(string[] args)
        {
            ReadWatchdogConfig rwc = new ReadWatchdogConfig();
            rwc.performActions();
        }

        public void showErrorMessage(LJM.LJMException e)
        {
            Console.Out.WriteLine("LJMException: " + e.ToString());
            Console.Out.WriteLine(e.StackTrace);
        }

        public void performActions()
        {
            int handle = 0;
            int devType = 0;
            int conType = 0;
            int serNum = 0;
            int ipAddr = 0;
            int port = 0;
            int maxBytesPerMB = 0;
            string ipAddrStr = "";

            try
            {
                //Open first found LabJack
                LJM.OpenS("ANY", "ANY", "ANY", ref handle);
                //devType = LJM.CONSTANTS.dtANY; //Any device type
                //conType = LJM.CONSTANTS.ctANY; //Any connection type
                //LJM.Open(devType, conType, "ANY", ref handle);

                LJM.GetHandleInfo(handle, ref devType, ref conType, ref serNum, ref ipAddr, ref port, ref maxBytesPerMB);
                LJM.NumberToIP(ipAddr, ref ipAddrStr);
                Console.WriteLine("Opened a LabJack with Device type: " + devType + ", Connection type: " + conType + ",");
                Console.WriteLine("Serial number: " + serNum + ", IP address: " + ipAddrStr + ", Port: " + port + ",");
                Console.WriteLine("Max bytes per MB: " + maxBytesPerMB);

                //Setup and call eReadNames to read the Watchdog configuration
                //from the LabJack.
                string[] aNames = new string[] {
                    "WATCHDOG_ENABLE_DEFAULT", "WATCHDOG_ADVANCED_DEFAULT",
                    "WATCHDOG_TIMEOUT_S_DEFAULT", "WATCHDOG_STARTUP_DELAY_S_DEFAULT",
                    "WATCHDOG_STRICT_ENABLE_DEFAULT", "WATCHDOG_STRICT_KEY_DEFAULT",
                    "WATCHDOG_RESET_ENABLE_DEFAULT", "WATCHDOG_DIO_ENABLE_DEFAULT",
                    "WATCHDOG_DIO_STATE_DEFAULT", "WATCHDOG_DIO_DIRECTION_DEFAULT",
                    "WATCHDOG_DIO_INHIBIT_DEFAULT", "WATCHDOG_DAC0_ENABLE_DEFAULT",
                    "WATCHDOG_DAC0_DEFAULT", "WATCHDOG_DAC1_ENABLE_DEFAULT",
                    "WATCHDOG_DAC1_DEFAULT"};
                double[] aValues = new double[aNames.Length];
                int numFrames = aNames.Length;
                int errAddr = -1;
                LJM.eReadNames(handle, numFrames, aNames, aValues, ref errAddr);

                Console.WriteLine("\nWatchdog configuration:");
                for(int i = 0; i < numFrames; i++)
                    Console.WriteLine("    " + aNames[i] + " : " + aValues[i]);
            }
            catch (LJM.LJMException e)
            {
                showErrorMessage(e);
            }

            LJM.CloseAll(); //Close all handles

            Console.WriteLine("\nDone.\nPress the enter key to exit.");
            Console.ReadLine(); // Pause for user
        }
    }
}

File: VB/Watchdog/ReadWatchdogConfig/ReadWatchdogConfig.vb
'------------------------------------------------------------------------------
' ReadWatchdogConfig.vb
'
' Demonstrates how to read the Watchdog configuration.
'
' support@labjack.com
'------------------------------------------------------------------------------
Option Explicit On

Imports LabJack

Module ReadWatchdogConfig

    Sub showErrorMessage(ByVal e As LJM.LJMException)
        Console.WriteLine("LJMException: " & e.ToString)
        Console.WriteLine(e.StackTrace)
    End Sub

    Sub displayHandleInfo(ByVal handle As Integer)
        Dim devType As Integer
        Dim conType As Integer
        Dim serNum As Integer
        Dim ipAddr As Integer
        Dim port As Integer
        Dim maxBytesPerMB As Integer
        Dim ipAddrStr As String = ""

        LJM.GetHandleInfo(handle, devType, conType, serNum, ipAddr, port, _
                          maxBytesPerMB)
        LJM.NumberToIP(ipAddr, ipAddrStr)
        Console.WriteLine("Opened a LabJack with Device type: " & devType & _
                          ", Connection type: " & conType & ",")
        Console.WriteLine("Serial number: " & serNum & ", IP address: " & _
                          ipAddrStr & ", Port: " & port & ",")
        Console.WriteLine("Max bytes per MB: " & maxBytesPerMB)
    End Sub

    Sub Main()
        Dim handle As Integer
        Dim numFrames As Integer
        Dim aNames() As String
        Dim aValues() As Double
        Dim errAddr As Integer = -1

        Try
            ' Open first found LabJack
            LJM.OpenS("ANY", "ANY", "ANY", handle)
            'LJM.Open(LJM.CONSTANTS.dtANY, LJM.CONSTANTS.ctANY, "ANY", handle)

            displayHandleInfo(handle)

            ' Setup and call eReadNames to read the Watchdog configuration from
            ' the LabJack.
            numFrames = 15
            ReDim aNames(numFrames - 1)
            aNames(0) = "WATCHDOG_ENABLE_DEFAULT"
            aNames(1) = "WATCHDOG_ADVANCED_DEFAULT"
            aNames(2) = "WATCHDOG_TIMEOUT_S_DEFAULT"
            aNames(3) = "WATCHDOG_STARTUP_DELAY_S_DEFAULT"
            aNames(4) = "WATCHDOG_STRICT_ENABLE_DEFAULT"
            aNames(5) = "WATCHDOG_STRICT_KEY_DEFAULT"
            aNames(6) = "WATCHDOG_RESET_ENABLE_DEFAULT"
            aNames(7) = "WATCHDOG_DIO_ENABLE_DEFAULT"
            aNames(8) = "WATCHDOG_DIO_STATE_DEFAULT"
            aNames(9) = "WATCHDOG_DIO_DIRECTION_DEFAULT"
            aNames(10) = "WATCHDOG_DIO_INHIBIT_DEFAULT"
            aNames(11) = "WATCHDOG_DAC0_ENABLE_DEFAULT"
            aNames(12) = "WATCHDOG_DAC0_DEFAULT"
            aNames(13) = "WATCHDOG_DAC1_ENABLE_DEFAULT"
            aNames(14) = "WATCHDOG_DAC1_DEFAULT"
            ReDim aValues(numFrames)
            LJM.eReadNames(handle, numFrames, aNames, aValues, errAddr)

            Console.WriteLine("")
            Console.WriteLine("Watchdog configuration:")
            For i = 0 To numFrames - 1
                Console.WriteLine("    " & aNames(i) & " : " & aValues(i))
            Next
        Catch ljme As LJM.LJMException
            showErrorMessage(ljme)
        End Try

        LJM.CloseAll() ' Close all handles

        Console.WriteLine("")
        Console.WriteLine("Done.")
        Console.WriteLine("Press the enter key to exit.")
        Console.ReadLine() ' Pause for user
    End Sub

End Module
